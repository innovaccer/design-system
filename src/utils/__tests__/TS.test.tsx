import * as React from 'react';
import * as DS from '../../index';
import { render } from '@testing-library/react';

// const noop = () => {};
const string = 'Hello World!';
const dummyComp = <DS.Text>{string}</DS.Text>;

export const TS = () => {
  return (
    <>
      <DS.Avatar />
      <DS.Badge>Badge</DS.Badge>
      <DS.Bold>{string}</DS.Bold>
      <DS.Button />
      <DS.Italic>{string}</DS.Italic>
      <DS.Link>{string}</DS.Link>
      <DS.Paragraph>{string}</DS.Paragraph>
      <DS.Pills>{string}</DS.Pills>
      <DS.Popover trigger={dummyComp}>{dummyComp}</DS.Popover>
      <DS.Strikethrough>{string}</DS.Strikethrough>
      <DS.Underline>{string}</DS.Underline>
      <DS.Tooltip text={string}>{dummyComp}</DS.Tooltip>
      <DS.Icon>{string}</DS.Icon>
      {/* <DS.AvatarGroup list={[]} />
      <DS.Backdrop open={false} />
      <DS.Breadcrumbs list={[]} onClick={noop} />
      <DS.Card>{dummyComp}</DS.Card>
      <DS.Chip name={string} label={string} />
      <DS.ChipGroup list={[]} />
      <DS.Checkbox />
      <DS.Column />
      <DS.DatePicker />
      <DS.Dropdown />
      <DS.EditableDropdown />
      <DS.EditableInput />
      <DS.Heading>{string}</DS.Heading>
      <DS.Input />
      <DS.InputMask mask={[]} />
      <DS.Label>{string}</DS.Label>
      <DS.Caption>{string}</DS.Caption>
      <DS.Legend>{string}</DS.Legend>
      <DS.Link>{string}</DS.Link>
      <DS.Message>{string}</DS.Message>
      <DS.MetaList list={[]} />
      <DS.OutsideClick onOutsideClick={noop}>{dummyComp}</DS.OutsideClick>
      <DS.ProgressBar value={0} />
      <DS.Radio name={string} value={string} />
      <DS.Row />
      <DS.StatusHint>{string}</DS.StatusHint>
      <DS.Spinner />
      <DS.Slider />
      <DS.RangeSlider />
      <DS.Subheading>{string}</DS.Subheading>
      <DS.Switch />
      <DS.Text>{string}</DS.Text>
      <DS.Textarea />
      <DS.Toast title={string} />
      <DS.Dialog
        open={false}
        onClose={noop}
        heading={string}
        primaryButtonLabel={string}
        primaryButtonCallback={noop}
        secondaryButtonLabel={string}
        secondaryButtonCallback={noop}
      />
      <DS.Modal open={false} />
      <DS.ModalHeader heading={string} onClose={noop} />
      <DS.ModalFooter>{dummyComp}</DS.ModalFooter>
      <DS.ModalBody>{dummyComp}</DS.ModalBody>
      <DS.ModalDescription />
      <DS.Pagination onPageChange={noop} />
      <DS.Placeholder />
      <DS.PlaceholderParagraph />
      <DS.ProgressRing value={0} />
      <DS.Stepper steps={[]} />
      <DS.DateRangePicker />
      <DS.TabsWrapper>{dummyComp}</DS.TabsWrapper>
      <DS.Tab label={dummyComp} />
      <DS.Grid totalRecords={50} />
      <DS.List />
      <DS.Table />
      <DS.Navigation menus={[]} />
      <DS.PageHeader title={string} />
      <DS.EditableChipInput placeholder="" /> */}
    </>
  );
};

beforeAll(() => {
  Date.now = () => new Date('2021-06-14').getTime();
});

describe('TS', () => {
  it('renders children', () => {
    const { asFragment } = render(TS());
    expect(asFragment()).toMatchSnapshot();
  });
});
