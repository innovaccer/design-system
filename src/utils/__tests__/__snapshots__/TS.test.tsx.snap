// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TS renders children 1`] = `
<DocumentFragment>
  <span
    class="Mds-Avatar Mds-Avatar--regular Mds-Avatar--secondary Mds-Avatar--disabled Mds-overflow-hidden"
    data-test="DesignSystem-Avatar"
  >
    <i
      class="material-icons material-icons-round Mds-Icon Mds-Icon--white Mds-Avatar-content--secondary"
      data-test="DesignSystem-Avatar--Icon"
      role="button"
      style="font-size: 16px; width: 16px;"
    >
      person_round
    </i>
  </span>
  <span
    class="Mds-Badge Mds-Badge--secondary"
    data-test="DesignSystem-Badge"
    role="status"
  >
    Badge
  </span>
  <b
    data-test="DesignSystem-Bold"
  >
    Hello World!
  </b>
  <button
    class="Mds-Button Mds-Button--regular Mds-Button--regularSquare Mds-Button--basic"
    data-test="DesignSystem-Button"
    tabindex="0"
  />
  <i
    data-test="DesignSystem-Italic"
  >
    Hello World!
  </i>
  <a
    aria-disabled="false"
    class="Mds-Link Mds-Link--regular Mds-Link--default"
    data-test="DesignSystem-Link"
  >
    Hello World!
  </a>
  <p
    class="Mds-Text Mds-Text--default"
    data-test="DesignSystem-Paragraph"
  >
    Hello World!
  </p>
  <span
    class="Mds-Pills Mds-Badge--secondary"
    data-test="DesignSystem-Pills"
    role="status"
  >
    Hello World!
  </span>
  <span
    aria-controls="floating-ui-1"
    aria-expanded="true"
    aria-haspopup="dialog"
    class="Mds-Text Mds-Text--default Mds-Text--regular"
    data-test="DesignSystem-Text"
  >
    Hello World!
  </span>
  <style />
  <s
    data-test="DesignSystem-Strikethrough"
  >
    Hello World!
  </s>
  <u
    data-test="DesignSystem-Underline"
  >
    Hello World!
  </u>
  <span
    aria-describedby="floating-ui-4"
    class="Mds-Text Mds-Text--default Mds-Text--regular"
    data-test="DesignSystem-Text"
  >
    Hello World!
  </span>
  <style />
  <i
    class="material-icons material-icons-round Mds-Icon"
    role="button"
    style="font-size: 16px; width: 16px;"
  >
    Hello World!_round
  </i>
</DocumentFragment>
`;
